src/bootstrap/src/utils/exec.rs

```diff
@@ -545,6 +607,7 @@ impl ExecutionContext {
         {
             command.mark_as_executed();
             self.verbose(|| println!("Cache hit: {command:?}"));
+            self.profiler.record(cache_key.unwrap(), Duration::from_secs(0));
```

Comment:
We should represent this in a proper way, e.g. store a `Vec` of execution traces, where each trace is either a cache hit, or an actual execution with a duration.